test/parallel/test-fs-stream-fs-options.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures');
+const path = require('path');
+const fs = require('fs');
+const assert = require('assert');
+
+const tmpdir = require('../common/tmpdir');
+tmpdir.refresh();
+
+const writeStreamOptions = ['open', 'close', 'write', 'writev'];
+const readStreamOptions = ['open', 'close', 'read'];
+
+{
+  const file = path.join(tmpdir.path, 'write-end-test0.txt');
+
+  writeStreamOptions.forEach((fn) => {
+    const opts = {
+      fs: {
+        [fn]: null
+      }
+    };
+    assert.throws(
+      () => fs.createWriteStream(file, opts), { code: 'ERR_INVALID_ARG_TYPE' }
```

Comment:
Would be nice to include a custom message including the method being executed, so if one fails it will be easier to identify, also, could be possible to validate the Error message too

Reply:
Addressed :) Thanks!